src/main/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcher.java

```diff
@@ -17,22 +21,33 @@
 
 public class CompositeSearchBasedFetcher implements SearchBasedFetcher {
 
-    public static final String FETCHER_NAME = "SearchAll";
+    public static final String FETCHER_NAME = "SearchSelected";
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CompositeSearchBasedFetcher.class);
 
-    private final Set<SearchBasedFetcher> fetchers;
+    private static Set<SearchBasedFetcher> fetchers;
     private final int maximumNumberOfReturnedResults;
 
-    public CompositeSearchBasedFetcher(Set<SearchBasedFetcher> searchBasedFetchers, int maximumNumberOfReturnedResults)
+    public CompositeSearchBasedFetcher(Set<SearchBasedFetcher> searchBasedFetchers, ImporterPreferences importerPreferences, int maximumNumberOfReturnedResults)
             throws IllegalArgumentException {
         if (searchBasedFetchers == null) {
             throw new IllegalArgumentException("The set of searchBasedFetchers must not be null!");
         }
-        // Remove the Composite Fetcher instance from its own fetcher set to prevent a StackOverflow
-        this.fetchers = searchBasedFetchers.stream()
-                                           .filter(searchBasedFetcher -> searchBasedFetcher != this)
-                                           .collect(Collectors.toSet());
+
+        ListProperty<String> catalogs = importerPreferences.getCatalogs();
```

Comment:
Can you rewrite the code so that an empty list is returned if not set? `null` is not good in modern Java.

After checking the code, `null` should never happen?